addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3000,11 +3000,12 @@ var SnippetsMenu = Widget.extend({
     _checkEditorToolbarVisibility: function (e) {
         const $toolbarContainer = this.$('#o_we_editor_toolbar_container');
         const docSelection = document.getSelection();
+        const $eventTarget = e ? $(e.target) : $();
         const $currentSelectionTarget = docSelection && docSelection.rangeCount > 0 ? $(docSelection.getRangeAt(0).commonAncestorContainer) : $();
-        // Do not  toggle visibility if the target is inside the toolbar ( eg. during link edition).
-        if ($currentSelectionTarget.parents('#toolbar').length ||
-            (e && $(e.target).parents('#toolbar').length)
-        ) {
+        const isClickOnDialog = $eventTarget.parents('.o_we_existing_attachments').length > 0 || $eventTarget.closest('.o_technical_modal').length > 0;
```

Comment:
When we click on an already imported file, the dialog is closed before this function is executed. Thus, we need another way to tell whether the click was on the dialog or not. `o_we_existing_attachments` doesn't seem indeed to be the best way, I figured the `$currentSelectionTarget` was a div with the `o_dialog_container` class  in this case and I used this information instead of `o_we_existing_attachments`. I think it's better now :  

![image](https://user-images.githubusercontent.com/48757558/126290781-74d83d5d-fc12-4e32-9361-a92bc3912a8c.png)